azure/scope/machinepoolmachine.go

```diff
@@ -249,6 +249,20 @@ func (s *MachinePoolMachineScope) ProviderID() string {
 	return s.AzureMachinePoolMachine.Spec.ProviderID
 }
 
+// TODO: does this need to be manually called in AMPM controller?
```

Comment:
> does this need to be manually called in AMPM controller?

I'm not sure what "manually" implies, but yes, this func will need to be invoked somewhere. Perhaps, add a `defer scope.Close()` in the controller and add a call to `PatchObject` in the `Close` func.

Reply:
Yep that makes sense. I just wanted to double check that we didn't need to call PatchObject() at the beginning.